stream-video-android-core/src/test/kotlin/io/getstream/video/android/core/stories/RingTest.kt

```diff
@@ -60,8 +65,37 @@ class RingTest : IntegrationTestBase() {
     }
 
     @Test
+    fun `Outgoing call is automatically cancelled`() = runTest {
+        client.state._ringingCall.value = null
+
+        val call = client.call("default", randomUUID())
+        call.create(ring = true)
+
+        // We are in outgoing state
+        assertThat(call.state.ringingState.value).isInstanceOf(RingingState.Outgoing::class.java)
+
+        // We wait until cancel time passes
+        val cancelTime = call.state.settings.value?.ring?.autoCancelTimeoutMs?.toLong() ?: 0
+        advanceTimeBy(cancelTime + 50)
+
+        // This is needed because we need to wait for call.reject() internally to finish
+        // Is there a way we could wait for the suspend function?
+        runBlocking {
+            delay(500)
+        }
+
+        // We verify that the call doesn't ring anymore and the state is back to Idle
+        assertThat(client.state.ringingCall.value).isNull()
+        assertThat(call.state.ringingState.value is RingingState.Idle)
```

Comment:
We should use [turbine](https://github.com/cashapp/turbine) to wait to get the emitted flow values instead of delaying with static durations.

Reply:
Yes, but in this is quite problematic - I was not able to do it yesterday, I will continue today. The StateFlow in this case is emitting multiple values as it progresses through the call state. Is there an easy way to wait for a specific StateFlow value? (or maybe verify that the StateFlow emits a specific sequence of values).

I was not able to remove the delay because internally our `wrapAPICall` is a `suspend` function and it's a real API call to our endpoints. We use a `UnconfinedTestDispatcher` in our `DispatcherRule` class therefore the execution order of queued coroutines is not guaranteed. And I don't yet know how we can wait for the API call to finish until we proceed.

Reply:
Ok, I fixed this. I used `waitForNextEvent<CallRejectedEvent>()` - so the code waits until the call is automatically rejected by our internal implementation. I had to modify our `waitForNextEvent` implementation because it was only advancing the virtual time while waiting for events - this doesn't work if we do real API calls in unit tests. So I added ` withContext(Dispatchers.Default.limitedParallelism(1))` inside `waitForNextEvent`.